File: scripts tienda_juguetes/juguetes_bd.py
import mysql.connector
from faker import Faker
import random
from datetime import datetime, timedelta

# Conexión a la base de datos
mydb = mysql.connector.connect(
  host="127.0.0.1",
  user="root",
  password="",
  database="juguetes"
)

mycursor = mydb.cursor()
fake = Faker()

# Función para generar datos falsos para la tabla juguete
def generate_juguete():
    nombre = fake.word()
    descripcion = fake.text()
    categoria = fake.random_element(elements=('Juguetes para niños', 'Juguetes para niñas', 'Juguetes educativos', 'Juguetes de exterior'))
    precio = round(random.uniform(5, 1000), 2)
    cantidad_disponible = random.randint(1, 100)
    fecha_ingreso = fake.date_between(start_date='-2y', end_date='today')
    return (nombre, descripcion, categoria, precio, cantidad_disponible, fecha_ingreso)

# Función para generar datos falsos para la tabla proveedores
def generate_proveedores(juguete_id):
    nombre = fake.company()
    descripcione = fake.text()
    precio = round(random.uniform(1, 100), 2)
    cantidad_disponible = random.randint(1, 50)
    return (nombre, descripcione, juguete_id, precio, cantidad_disponible)

# Función para generar datos falsos para la tabla coleccion
def generate_coleccion(juguete_id):
    nombre = fake.word()
    fecha_salida = fake.date_between(start_date='today', end_date='+2y')
    return (nombre, fecha_salida, juguete_id)

# Generar datos falsos y llenar la tabla juguete
for _ in range(500):
    juguete_data = generate_juguete()
    sql = "INSERT INTO juguete (nombre, descripcion, categoria, precio, cantidad_disponible, fecha_ingreso) VALUES (%s, %s, %s, %s, %s, %s)"
    mycursor.execute(sql, juguete_data)
    juguete_id = mycursor.lastrowid
    mydb.commit()
    
    # Llenar la tabla proveedores
    proveedores_data = generate_proveedores(juguete_id)
    sql = "INSERT INTO proveedores (nombre, descripcione, producto, precio, cantidad_disponible) VALUES (%s, %s, %s, %s, %s)"
    mycursor.execute(sql, proveedores_data)
    mydb.commit()

    # Llenar la tabla coleccion
    coleccion_data = generate_coleccion(juguete_id)
    sql = "INSERT INTO coleccion (nombre, fecha_salida, producto) VALUES (%s, %s, %s)"
    mycursor.execute(sql, coleccion_data)
    mydb.commit()

print("Datos insertados correctamente.")

File: scripts tienda_juguetes/juguetes_bd.js
const faker = require('faker');
const mysql = require('mysql');

// Configuración de conexión a la base de datos
const connection = createConnection({
  host: 'localhost',
  user: 'root',
  password: '',
  database: 'juguetes'
});

// Conectar a la base de datos
connection.connect((err) => {
  if (err) throw err;
  console.log('Conexión establecida con la base de datos');
});

// Función para generar datos de juguetes
function generateJuguete() {
  return {
    nombre: commerce.productName(),
    descripcion: lorem.paragraph(),
    categoria: random.arrayElement(['Muñecas', 'Construcción', 'Vehículos', 'Puzzles']),
    precio: parseFloat(commerce.price()),
    cantidad_disponible: datatype.number({ min: 1, max: 100 }),
    fecha_ingreso: date.between('2020-01-01', '2024-05-01').toISOString().split('T')[0]
  };
}

// Función para generar datos de proveedores
function generateProveedor() {
  return {
    nombre: company.companyName(),
    descripcione: lorem.paragraph(),
    producto: datatype.number({ min: 1, max: 100 }),
    precio: parseFloat(commerce.price()),
    cantidad_disponible: datatype.number({ min: 1, max: 100 })
  };
}

// Función para generar datos de colecciones
function generateColeccion() {
  return {
    nombre: lorem.words(),
    fecha_salida: date.between('2020-01-01', '2024-05-01').toISOString().split('T')[0],
    producto: datatype.number({ min: 1, max: 100 })
  };
}

// Insertar datos de juguetes
for (let i = 0; i < 1000; i++) {
  const juguete = generateJuguete();
  connection.query('INSERT INTO juguete SET ?', juguete, (err, result) => {
    if (err) throw err;
    console.log('Juguete insertado:', result.insertId);
  });
}

// Insertar datos de proveedores
for (let i = 0; i < 200; i++) {
  const proveedor = generateProveedor();
  connection.query('INSERT INTO proveedores SET ?', proveedor, (err, result) => {
    if (err) throw err;
    console.log('Proveedor insertado:', result.insertId);
  });
}

// Insertar datos de colecciones
for (let i = 0; i < 300; i++) {
  const coleccion = generateColeccion();
  connection.query('INSERT INTO coleccion SET ?', coleccion, (err, result) => {
    if (err) throw err;
    console.log('Coleccion insertada:', result.insertId);
  });
}

// Cerrar conexión a la base de datos
connection.end();
